src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/RemoveAzureRmCdnCustomDomain.cs

```diff
@@ -63,48 +67,38 @@ public override void ExecuteCmdlet()
                 CustomDomainName = CdnCustomDomain.Name;
             }
 
-            try
-            {
-                CdnManagementClient.CustomDomains.GetWithHttpMessagesAsync(
-                    CustomDomainName, 
-                    EndpointName, 
-                    ProfileName, 
-                    ResourceGroupName).Wait();
-            }
-            catch (AggregateException exception)
-            {
-                var errorResponseException = exception.InnerException as ErrorResponseException;
-                if (errorResponseException == null)
-                {
-                    throw;
-                }
 
-                if (errorResponseException.Response.StatusCode.Equals(HttpStatusCode.NotFound))
-                {
-                    throw new PSArgumentException(string.Format(Resources.Error_DeleteNonExistingCustomDomain,
-                        CustomDomainName,
-                        EndpointName,
-                        ProfileName,
-                        ResourceGroupName));
-                }
+            var existingCustomDomain = CdnManagementClient.CustomDomains.ListByEndpoint(EndpointName, ProfileName, ResourceGroupName)
+                .Where(cd => cd.Name.ToLower() == CustomDomainName.ToLower())
+                .FirstOrDefault();
+
+            if (existingCustomDomain == null)
+            {
+                throw new PSArgumentException(string.Format(Resources.Error_DeleteNonExistingCustomDomain,
+                    CustomDomainName,
+                    EndpointName,
+                    ProfileName,
+                    ResourceGroupName));
             }
 
-            if (!ShouldProcess(string.Format(
+           if (Force || ShouldProcess(string.Format(
```

Comment:
This is not the pattern for PowerShell Confirmation.  Force should only be used with a ShouldContinue prompt, after a ShouldProcess check.  Force is really only for specially destructive actions, you are already gestting a ShouldProcess prompt in most cases beacuase your ConfirmImpact is High.

Please remove the Force prompt and revert this conditional change